src/main/java/randoop/operation/MethodCall.java

```diff
@@ -92,45 +100,102 @@ public void appendCode(
-            .append(")");
+    if (isAccessible) {
+      if (isStatic()) {
+        // In the generated Java code, the "receiver" (before the method name) is the class name.
+        sb.append(declaringType.getCanonicalName().replace('$', '.'));
       } else {
-        sb.append(receiverVar);
+        // In the generated Java code, the receiver is an expression.
+        Type receiverFormalType = inputTypes.get(0);
+        if (receiverFormalType.isPrimitive()) {
+          sb.append("((")
+              .append(receiverFormalType.getFqName())
+              .append(")")
+              .append(receiverVar)
+              .append(")");
+        } else {
+          sb.append(receiverVar);
+        }
       }
-    }
 
-    sb.append(".");
-    sb.append(methodName);
+      sb.append(".");
+      sb.append(methodName);
+    } else {
+      if (!outputType.isVoid()) {
+        // Cast because the return type of `invoke()` is Object.
+        sb.append("(").append(outputType.getFqName()).append(") ");
+      }
+      sb.append(methodVar).append(".").append("invoke");
+    }
 
     StringJoiner arguments = new StringJoiner(", ", "(", ")");
-    int startIndex = (isStatic() ? 0 : 1);
+    // In a reflective call, a receiver is always passed (even if it's null).
+    int startIndex = !isAccessible || isStatic() ? 0 : 1;
     for (int i = startIndex; i < inputVars.size(); i++) {
-      // CASTING.
-      String cast;
-      if (inputVars.get(i).getType().equals(inputTypes.get(i))) {
-        cast = "";
+      if (i == 0 && !isAccessible && isStatic()) {
+        // There is no harm to passing inputVars.get(0), but pass
+        // null to emphasize that the first (receiver) argument is ignored.
+        sb.append("null");
       } else {
-        // Cast if the argument and formal parameter types are not identical.
-        cast = "(" + inputTypes.get(i).getFqName() + ") ";
+        // CASTING.
+        String cast;
+        if (inputVars.get(i).getType().equals(inputTypes.get(i))) {
+          cast = "";
+        } else {
+          // Cast if the argument and formal parameter types are not identical.
+          cast = "(" + inputTypes.get(i).getFqName() + ") ";
+        }
+        String param = getArgumentString(inputVars.get(i));
+        arguments.add(cast + param);
       }
-      String param = getArgumentString(inputVars.get(i));
-      arguments.add(cast + param);
     }
     sb.append(arguments.toString());
   }
 
+  /**
+   * Constructs a unique variable name for the method object associated with this {@link
+   * MethodCall}. The variable name is formed by appending the method name and the simple names of
+   * the parameter types, separated by underscores. For array types, "Array" is appended instead of
+   * "[]".
+   *
+   * @return the unique variable name
+   */
+  private String getVariableNameForMethodObject() {
+    StringBuilder signature = new StringBuilder();
+    // Append method name
+    signature.append(getMethod().getName());
```

Comment:
This isn't necessarily unique.  There might be a method named "foo" in multiple classes (especially with overriding).  Should you add the class name?